v2/php/adsense-sample.php

```diff
@@ -0,0 +1,203 @@
+<?php
+/*
+ * Copyright 2021 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * A sample application that runs multiple requests against the AdSense
+ * Management API. These include:
+ * <ul>
+ * <li>Listing all AdSense accounts for a user</li>
+ * <li>Listing the sub-account tree for an account</li>
+ * <li>Listing all ad clients for an account</li>
+ * <li>Listing all ad units for an ad client</li>
+ * <li>Listing all custom channels for an ad unit</li>
+ * <li>Listing all custom channels for an ad client</li>
+ * <li>Listing all ad units for a custom channel</li>
+ * <li>Listing all URL channels for an ad client</li>
+ * <li>Running a report for an ad client</li>
+ * <li>Listing all saved reports for an account</li>
+ * <li>Running a saved report for an account</li>
+ * <li>Listing all alerts for an account</li>
+ * </ul>
+ *
+ */
+
+require_once 'templates/base.php';
+session_start();
+
+/************************************************
+  ATTENTION: Change this path to point to your vendor folder if your project
+  directory structure differs from this repository's!
+ ************************************************/
+require_once __DIR__ . '/vendor/autoload.php';
+
+// Autoload example classes.
+spl_autoload_register(function ($class_name) {
+  include 'examples/' . $class_name . '.php';
+});
+
+// Max results per page.
+define('MAX_LIST_PAGE_SIZE', 50, true);
+define('MAX_REPORT_PAGE_SIZE', 50, true);
+
+// Configure token storage on disk.
+// If you want to store refresh tokens in a local disk file, set this to true.
+define('STORE_ON_DISK', false, true);
+define('TOKEN_FILENAME', 'tokens.dat', true);
+
+// Set up authentication.
+$client = new Google_Client();
+$client->addScope('https://www.googleapis.com/auth/adsense.readonly');
+$client->setAccessType('offline');
+
+// Be sure to replace the contents of client_secrets.json with your developer
+// credentials.
+$client->setAuthConfig('client_secrets.json');
+
+// Create service.
+$service = new Google_Service_Adsense($client);
+
+// Logging out will remove the current access token and delete the token file
+// if a token file exists (i.e., STORE_ON_DISK is enabled).
+if (isset($_REQUEST['logout'])) {
+  unset($_SESSION['access_token']);
+  unlink(TOKEN_FILENAME);
+}
+
```

Comment:
Added ```// Configure redirect uri to return to main page once the authentication workflow has completed.
$redirectUri = 'http://' . $_SERVER['HTTP_HOST'] . $_SERVER['PHP_SELF'];
$client->setRedirectUri($redirectUri);` to enable redirect.

Reply:
Can you help me understand exactly where this is needed? In the logout conditional?